nixops_aws/resources/ec2_rds_dbinstance.py

```diff
@@ -260,14 +302,33 @@ def _to_boto_kwargs(self, attrs):
             "rds_dbinstance_instance_class": "instance_class",
             "rds_dbinstance_multi_az": "multi_az",
             "rds_dbinstance_security_groups": "security_groups",
+            "vpc_security_groups": "vpc_security_groups",
         }
-        return {attr_to_kwarg[attr]: attrs[attr] for attr in attrs.keys()}
+        result = {attr_to_kwarg[attr]: attrs[attr] for attr in attrs.keys()}
+
+        # If the user has specified VPC security groups, and
+        # have left security groups as the default list of "default",
+        # they probably don't use the default security group --
+        # and it is incompatible with VPC security groups anyway.
+        if (
+            "vpc_security_groups" in result
+            and "security_groups" in result
+            and len(result["vpc_security_groups"]) > 0
+            and list(result["security_groups"]) == ["default"]
```

Comment:
Testing tuple equality in this case seems sounder than list equality IMHO

Reply:
Why is that? I don't think I understand the difference.

Reply:
At the right, you're instantiating a `list`, so a full-fledged mutable object in Python, which is in fact, mutable.

```python
>>> def f(L):
...    return list(L) == ["default"]
... 
>>> dis.dis(f)
  2           0 LOAD_GLOBAL              0 (list)
              2 LOAD_FAST                0 (L)
              4 CALL_FUNCTION            1
              6 LOAD_CONST               1 ('default')
              8 BUILD_LIST               1
             10 COMPARE_OP               2 (==)
             12 RETURN_VALUE
```
Here's the "bytecode" using the `dis` module.

Now, with the `tuple` version:

```python
>>> def g(L):
...     return tuple(L) == ("default",)
... 
>>> dis.dis(g)
  2           0 LOAD_GLOBAL              0 (tuple)
              2 LOAD_FAST                0 (L)
              4 CALL_FUNCTION            1
              6 LOAD_CONST               1 (('default',))
              8 COMPARE_OP               2 (==)
             10 RETURN_VALUE
```
So there is an actual difference, you're saving a list construction (because tuples are immutable and have nice properties Python can use to optimize things around).

**But**, micro-performance is not the only reason to use it, for more niceties, please refer to https://stackoverflow.com/questions/68630/are-tuples-more-efficient-than-lists-in-python/22140115#22140115

Also, IMHO, when I review stuff, I keep track of "singleton list" because they often mean to me that you're writing an imperative mutable algorithm or something like this (which is clearly not the case here AFAIK)

Reply:
Addressed, thanks!